ruby/Rakefile

```
@@ -0,0 +1,186 @@
+
+  release = find_or_create_release(client, repo, FluentTools::VERSION)
+  asset = upload_release_asset(client, release, binary_file)
+
+  puts "✅ Successfully uploaded #{binary_file} to GitHub release!"
+  puts "🔗 Download URL: #{asset.browser_download_url}"
+
+  # Clean up local release binary
+  FileUtils.rm(binary_file)
+end
+
+desc 'Build binaries for all supported platforms and upload to GitHub releases'
+task :release_all_platforms do
+  require_project_deps
+
+  FluentTools::Utils::SUPPORTED_PLATFORMS.each do |platform|
+    puts "\n🔨 Building and releasing for #{platform}..."
+    Rake::Task[:release_binary].reenable
+    Rake::Task[:release_binary].invoke(platform)
+  end
+
+  puts "\n🎉 All platform binaries built and uploaded successfully!"
+end
+
+desc 'Clean all build artifacts'
+task :clean do
+  FileUtils.rm_rf(%w[bin pkg])
+  system("cd #{project_root} && make clean")
+  puts '🧹 Cleaned all build artifacts'
+end
+
+# Helper methods
+def project_root
+  @project_root ||= File.expand_path('..', __dir__)
+end
+
+def require_project_deps
+  require_relative 'lib/fluent_tools/version'
+  require_relative 'lib/fluent_tools/utils'
+end
+
+def build_rust_binary(platform)
+  FluentTools::Utils.validate_platform!(platform)
+  puts "🔨 Building Rust binary for platform: #{platform}"
+
+  system("cd #{project_root} && make build_platform PLATFORM=#{platform}") || raise('Failed to build Rust binary')
+end
+
+def get_binary_paths(platform, target_dir)
+  require_project_deps
+
+  raise 'Platform is required' unless platform
+  raise 'Target directory is required' unless target_dir
+
+  binary_extension = FluentTools::Utils.binary_extension(platform)
+  binary_name = FluentTools::Utils::BINARY_NAME
+
+  source_path = File.expand_path("../target/#{target_dir}/release/#{binary_name}#{binary_extension}", __dir__)
+
+  {
+    platform: platform,
+    extension: binary_extension,
+    source_path: source_path,
+    gem_binary_path: "bin/#{binary_name}#{binary_extension}",
+    release_binary_name: FluentTools::Utils.binary_name_for_platform(platform)
+  }
+end
+
+def copy_binary_for_gem(paths)
+  FileUtils.mkdir_p('bin')
+
+  unless File.exist?(paths[:source_path])
+    puts "❌ Source binary not found at #{paths[:source_path]}"
+    exit 1
+  end
+
+  FileUtils.cp(paths[:source_path], paths[:gem_binary_path])
+
+  # Make executable on Unix-like systems
+  File.chmod(0o755, paths[:gem_binary_path]) unless FluentTools::Utils.windows_platform?(paths[:platform])
+
+  puts "✅ Binary copied to #{paths[:gem_binary_path]}"
+  paths[:gem_binary_path]
+end
+
+def prepare_release_binary(paths)
+  unless File.exist?(paths[:source_path])
+    puts "❌ Built binary not found at #{paths[:source_path]}"
+    exit 1
+  end
+
+  FileUtils.cp(paths[:source_path], paths[:release_binary_name])
+  puts "📦 Prepared release binary: #{paths[:release_binary_name]}"
+  paths[:release_binary_name]
+end
+
+def find_or_create_release(client, repo, version)
```

Given all the mentions of "Release" in that `Rakefile` that could be misinterpreted to mean "Release build, as opposed to Dev/Debug build" instead of "GitHub Release", I think it'd be nice to make those task names more explicit:
```suggestion
def find_or_create_github_release(client, repo, version)
```

---

Agreed 👍 Updated on a62d0355cd2b4de20105c99f0ee5b270e2f59c58.